### PR TITLE
Add threaded websocket client

### PR DESCRIPTION
**Pull request by Numerlor**
Adds a thread safe websocket client that can dispatch messages to multiple subscribed consumers through queues. Example usage here https://github.com/Momentous-Muses/code-jam/blob/fd7eb93abb625390e2342296e3ae22639d88e949/client/websocket_client/client.py#L115-L134

New message classes are created as Message subclasses, e.g.
```py
@dataclass
class CustomMessage(websocket_client.Message):
    channel = "test_channel"
    type = "test_type"

    @classmethod
    def from_json_dict(cls, json_):
        return cls_created_from_json

    def to_json(self) -> str:
        return json.dumps(as_dict(self))
```
after the Message subclass is created, when a message with the given type is received on a channel, the class is created from the received json and sent to all queues on the given channel.

The uassage I have in mind is that the tkinter client will create the dispatcher, keep it around and for each queue it subscribes for (each channel will represent a different, separate part of the communication), it'll poll it with a function scheduled with `app.after`

### CODE REVIEW

**Review by c-gaskell — APPROVED** [review]

File: client/websocket_client/__init__.py
Comment (ooliver1) [inline client/websocket_client/__init__.py]:
```suggestion
__all__ = ("Message", "ResultEvent", "UNINITIALIZED", "WebsocketMessageDispatcher")
```
`__all__` does not need mutation

**Review by ooliver1 — CHANGES_REQUESTED** [review]
Otherwise, looks good!

**Comment by c-gaskell** [comment]
There's merge conflicts for poetry.lock - i'm not entirely sure how this file works, does anyone else know how this conflict should be resolved?

**Comment by Numerlor** [comment]
for lock file conflicts, it just has to be generated again; I've resolved it and brought the branch up to date with main

**Comment by c-gaskell** [comment]
Just waiting for the review from @ooliver1 before I can merge as he requested changes, once that's done I'm good to merge

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**
